common/lib/xmodule/xmodule/js/src/video/01_initialize.js

```diff
@@ -465,6 +448,54 @@ function (VideoPlayer, i18n, moment) {
         });
     }
 
+    function loadYoutubePlayer() {
+        console.log(
+            '[Video info]: Start player in YouTube mode.'
+        );
+
+        this.fetchMetadata();
+        this.parseSpeed();
+    }
+
+    function loadHtmlPlayer() {
+        console.log(
+            '[Video info]: YouTube returned an error for ' +
+            'video with id "' + this.id + '".'
+        );
+
+        // When the youtube link doesn't work for any reason
+        // (for example, the great firewall in china) any
+        // alternate sources should automatically play.
+        if (!_prepareHTML5Video(this)) {
+            console.log(
+                '[Video info]: Continue loading ' +
+                'YouTube video.'
+            );
+
+            // Non-YouTube sources were not found either.
+
+            this.el.find('.video-player div')
+                .removeClass('hidden');
+            this.el.find('.video-player h3')
+                .addClass('hidden');
+
+            // If in reality the timeout was to short, try to
+            // continue loading the YouTube video anyways.
+            this.fetchMetadata();
+            this.parseSpeed();
+        } else {
+            console.log(
+                '[Video info]: Change player mode to HTML5.'
+            );
+
+            // In-browser HTML5 player does not support quality
+            // control.
+            this.el.find('a.quality_control').hide();
+        }
+        _renderElements(this);
```

Comment:
This is not being called in `loadYoutubePlayer()` but does get called if `_prepareHTML5Video()` returns `false`.